### PR TITLE
Fix optional cycle-freshness compatibility path

### LINKED ISSUE #135: Updater still fails when optional cycle-freshness command is absent
## TL;DR

The `eva-v0.41.18.1` local canary still fails during Support KB install when upstream GBrain does not expose `gbrain sources cycle-freshness`. The first patch guarded the command with a help-text probe, but the live canary still reached the optional command and exited nonzero.

## Why this matters

This blocks fleet rollout even though the installed GBrain, OpenClaw plugin, Codex plugin, and Support KB all reached the correct version and loaded successfully. The `cycle-freshness` step is optional compatibility behavior; absence of that subcommand should never fail an install/update.

## Observed live failure

From the local exact-tag canary on `eva-v0.41.18.1`:

```text
+ /Users/lume/.bun/bin/gbrain sources cycle-freshness openclaw-support-kb off
Unknown sources subcommand: cycle-freshness
```

Immediately before that, the canary had succeeded through:

- `gbrain init` with `gbrain 0.41.18.1`
- OpenClaw plugin install and runtime inspect, plugin version `0.41.18.1`
- Codex plugin install
- Support KB update/status/sync/embed with `1101` source pages

## Proposed fix

Make `disable_support_kb_cycle_freshness_if_supported` run the optional command directly and classify the result:

```mermaid
flowchart TD
  A["Run optional cycle-freshness disable"] --> B{"Command exits 0?"}
  B -->|Yes| C["Continue"]
  B -->|No| D{"stderr says unknown cycle-freshness subcommand?"}
  D -->|Yes| E["Log skip and continue"]
  D -->|No| F["Fail updater with original error"]
```

This is safer than relying on help text and still preserves failure for real data/config errors.

## Acceptance criteria

- Updater does not fail when `gbrain sources cycle-freshness` is absent.
- Updater still fails if `cycle-freshness` exists but returns an unrelated error.
- Tests cover the live unknown-subcommand path with a fake `~/.bun/bin/gbrain`.
- Version bumps to a new patch tag, expected `0.41.18.2`.

### PR DESCRIPTION
**Pull request by 100yenadmin**
Closes #135.

## TL;DR

`eva-v0.41.18.1` proved the local canary can still fail on upstream GBrain installs that no longer expose `gbrain sources cycle-freshness`. This PR makes that optional compatibility step robust: if the command is missing, the updater logs a skip and continues; if the command exists but fails for a real reason, the updater still fails.

## Why

The Support KB refresh already succeeded locally before this optional step:

```mermaid
flowchart TD
  A["Install Eva Brain"] --> B["Install OpenClaw plugin"]
  B --> C["Install Codex plugin"]
  C --> D["Update Support KB"]
  D --> E["gbrain sync + embed openclaw-support-kb"]
  E --> F["Optional cycle-freshness disable"]
  F --> G["Doctor + health smoke"]
```

On current upstream GBrain, step `F` may not exist. That should not block local canary or fleet rollout because it is not required for indexing/search/plugin runtime. But we still want real errors from that command to stop the rollout if a future GBrain implements it and reports an actual data/config problem.

## What Changed

- Bump patch release metadata to `0.41.18.2`.
- Replace the help-text probe with direct optional command execution.
- Classify only this exact compatibility error as skippable:
  - `Unknown sources subcommand: cycle-freshness`
- Preserve fail-closed behavior for all other errors.
- Add focused updater contract tests with fake installed `~/.bun/bin/gbrain`:
  - missing `cycle-freshness` exits successfully and logs skip
  - unrelated `cycle-freshness` error exits nonzero

## Validation

Local focused only, from `/Volumes/LEXAR/repos/eva-brain-041182-hotfix`:

```bash
bash -n scripts/update-local-install.sh
bun test test/local-updater-contract.test.ts
bun test test/install-contract.test.ts
git diff --check
```

All passed. Full matrix should run in GitHub Actions.

## Release Note

After merge, tag `eva-v0.41.18.2` and use that exact tag for the local canary. Do not roll `.1` to fleet; it is superseded by this compatibility fix.

## Release Notes

* **Chores**
  * Released version 0.41.18.2 across all packages and plugins

* **Bug Fixes**
  * Enhanced Support KB installation reliability by improving error handling for optional features

* **Tests**
  * Added comprehensive test coverage for Support KB installation and edge case scenarios

[Review Change Stack](https://app.coderabbit.ai/change-stack/electricsheephq/eva-brain/pull/136?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: scripts/update-local-install.sh [hunk 1]
```diff
@@ -330,13 +330,25 @@ install_support_kb() {
...
+  if output="$("${cmd[@]}" 2>&1)"; then
+    printf '%s\n' "$output"
+    return
+  fi
+  if printf '%s\n' "$output" | grep -q 'Unknown sources subcommand: cycle-freshness'; then
     log "Skipping cycle-freshness disable; installed gbrain does not expose 'sources cycle-freshness'."
+    return
   fi
+  printf '%s\n' "$output" >&2
+  return 1
```
[the remark this answers is not in the document]
Comment (100yenadmin) [reply scripts/update-local-install.sh]: Fixed in `748672d4`. The compatibility classifier now normalizes output and requires an exact `Unknown sources subcommand: cycle-freshness` match before skipping; non-matching output still prints to stderr and returns failure. Test: `test/local-updater-contract.test.ts` now makes the negative fake-gbrain path include the compatibility phrase plus a real error, proving we stay fail-closed.
